src/components/Calculator.js

```diff
@@ -1,40 +1,139 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import Button from './Button';
 import './calculator.css';
+import calculate from '../logic/calculate';
 
-const Calculator = () => (
-  <div className="calculator-container">
-    <input type="text" value="0" />
-    <div className="buttons-aligner">
-      <Button buttonClass="mainClass" text="AC" />
-      <Button buttonClass="mainClass" text="+/-" />
-      <Button buttonClass="mainClass" text="%" />
-      <Button buttonClass="mainClass subclass" text="&divide;" />
-    </div>
-    <div className="buttons-aligner">
-      <Button buttonClass="mainClass" text="7" />
-      <Button buttonClass="mainClass" text="8" />
-      <Button buttonClass="mainClass" text="9" />
-      <Button buttonClass="mainClass subclass" text="&times;" />
-    </div>
-    <div className="buttons-aligner">
-      <Button buttonClass="mainClass" text="4" />
-      <Button buttonClass="mainClass" text="5" />
-      <Button buttonClass="mainClass" text="6" />
-      <Button buttonClass="mainClass subclass" text="-" />
-    </div>
-    <div className="buttons-aligner">
-      <Button buttonClass="mainClass" text="1" />
-      <Button buttonClass="mainClass" text="2" />
-      <Button buttonClass="mainClass" text="3" />
-      <Button buttonClass="mainClass subclass" text="+" />
-    </div>
-    <div className="buttons-aligner">
-      <Button buttonClass="mainClass subsidiary" text="0" />
-      <Button buttonClass="mainClass" text="." />
-      <Button buttonClass="mainClass subclass" text="=" />
-    </div>
-  </div>
-);
+const Calculator = () => {
+  const [screen, setScreen] = useState('0');
+  const [buttontype, setButtontype] = useState(null);
+  const [calculatoroperation, setCalculatoroperation] = useState(null);
+  const [operationnumber, setOperationnumber] = useState(null);
 
+  useEffect(() => {
+    // console.log(screen, operationnumber, calculatoroperation, buttontype);
+    const total = calculate(
+      {
+        total: screen,
+        next: operationnumber,
+        operation: calculatoroperation,
+      },
+      buttontype,
+    );
+    if (total.total) {
+      setScreen(total.total);
+    } else {
+      setScreen('0');
+    }
+  }, [buttontype, operationnumber]);
```

Comment:
You have done well so far , but the calculator doesn't work the way it should be as it doesn't display the neither operator nor the second operand which is bad for user experience as the user get"s confused if the button was clicked or not ,also the calculation is performed before we hit the equal button and another calculation is done after we click it which gives an incorrect result and the issue is that you call calculate each time a button was clicked, to fix this you can use one state with the object {total, next, operator} needed for calculate and and set this state with the value returned from calculate each time we hit a button since calculate handles all diffrent cases 

`const [data, setData] = useState({
    total: '',
    next: null,
    operation: null,
  });

  const handlClick = (e) => {
    setData((state) => ({
      ...state,
      ...calculate(state, e.target.name),
    }));
  };`